api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -360,6 +360,13 @@ message HttpConnectionManager {
   google.protobuf.Duration request_timeout = 28
       [(udpa.annotations.security).configure_for_untrusted_downstream = true];
 
+  // The amount of time that Envoy will wait for the headers to be received. The timer is activated
+  // when the request is initiated, and is disarmed when the last byte of the headers is sent
+  // upstream (i.e. all decoding filters have processed the headers). If not specified or set to 0,
+  // this timeout is disabled.
+  google.protobuf.Duration request_headers_timeout = 41
```

Comment:
Hm, this sounds like a different timeout than described in the issue.  I think issue is about the client sending headers (not hogging resources with a trickle attack).  Going through the filter chain seems to mix Envoy processing time with potentially malicious clients.  Did we want the latter as well?

Also, do we think this is a fixed timer long term, or do we hope it will evolve into the ranged timeout in the linked isssue?  If so I think we want to start with a different API and hide it until the range timeout is implemented.

I'm going to tag antonio for first pass here as I think he has a bit more context than I do.

Reply:
+1 I would not recommend making this involve filters at all. I think we want a timer that stats at stream init (1st header block), and ends after all blocks/continuations are complete?

Given ^ it might be easier to configure this at the codec/protocol level?

Reply:
Agreed, the upstreams bit was a mistake on my part. I think the existing implementation that disables the timer in onHeadersReceived does what we want.